pkg/cloudprovider/azure/object_store.go

```diff
@@ -146,22 +147,31 @@ func getStorageAccountKey(config map[string]string) (string, error) {
 		return "", errors.Wrap(err, "unable to get all required environment variables")
 	}
 
-	// 2. we need config["resourceGroup"], config["storageAccount"]
+	// 2. check whether a different subscription ID was set for backups in config["subscriptionId"]
+	var subscriptionId string
```

Comment:
I'd probably simplify this section to something like:

```go
subscriptionID := envVars[subscriptionIDEnvVar]
if val := config[subscriptionIdConfigKey]; val != "" {
	subscriptionID = val
}
```

Reply:
Sure, thanks. Just digging into the Go stuff.